### ISSUE TITLE
How do you change the logs format and level?

### ISSUE THREAD

**Issue by Drillan767** [lucaslorentz/caddy-docker-proxy#289]
Hello, I'd like to change the default configuration for the reverse-proxy, because it returns tons of stuff that we don't necessary require, while it's getting harder to spot where the warnings / error are.

I tried to add labels to the reverse proxy's docker-compose file, but it didn't work, even though the container started without trouble.

Here is what I'd like:

```
{
    log {
        format console
        level warning
   }
}
```

Here is how I tried to set my labels:

```
version: "3.7"
services:
  caddy:
    image: lucaslorentz/caddy-docker-proxy:ci-alpine
    ports:
      - 80:80
      - 443:443
    networks:
      - caddy
    environment:
      - CADDY_INGRESS_NETWORKS=caddy
    labels:
      caddy.log.output: console
      caddy.log.level: warning 
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - caddy_data:/data
    restart: unless-stopped

networks:
  caddy:
    external: true

volumes:
  caddy_data: {}
```

What am I doing wrong? And is this even possible? Thank you

**Comment by francislavoie** [comment]
What's in your logs for your `caddy` service? What does the generated Caddyfile look like?

**Comment by thechubbypanda** [comment]
This isn't working on the latest docker hub version as of now. 
```
version: "3.7"
services:
  caddy:
    image: lucaslorentz/caddy-docker-proxy:ci-alpine
    container_name: caddy
    restart: always
    ports:
      - 80:80
      - 443:443
      - 8448:8448
    environment:
      - CADDY_INGRESS_NETWORKS=caddy
    networks:
      - caddy
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - ./data:/data
    extra_hosts:
      - "host.docker.internal:host-gateway"
    labels:
      caddy: pluto.thechubbypanda.net
      caddy.reverse_proxy: "host.docker.internal:9090"
    labels:
      caddy.log.output: console
      caddy.log.level: warning

networks:
  caddy:
    external: true
```
Error:
```
{"level":"info","ts":1630438143.5122013,"logger":"docker-proxy","msg":"Process Caddyfile","logs":"[ERROR]  Removing invalid block: parsing caddyfile tokens for 'log': Caddyfile:4 - Error during parsing: getting module named 'caddy.logging.writers.console': module not registered: caddy.logging.writers.console\n{\n\tlog {\n\t\tlevel warning\n\t\toutput console\n\t}\n}\n\n"}
```

**Comment by francislavoie** [comment]
Right, so as per the docs, `console` is a _format_, not an _output_.

https://caddyserver.com/docs/caddyfile/directives/log

**Comment by Drillan767** [comment]
> What's in your logs for your `caddy` service?

My logs look like the following:

```
caddy_1  | {"level":"info","ts":1630449735.5444095,"logger":"docker-proxy","msg":"Skipping default Caddyfile because no path is set"}
caddy_1  | {"level":"info","ts":1630449735.5444677,"logger":"docker-proxy","msg":"Skipping swarm config caddyfiles because swarm is not available"}
caddy_1  | {"level":"info","ts":1630449735.5470812,"logger":"docker-proxy","msg":"Skipping swarm services because swarm is not available"}
```

I have tons and tons of lines like this, hence my question about filtering / displaying logs in a nicer way.

> What does the generated Caddyfile look like?

Where do I find the generated Caddyfile? I browsed the Caddy container for it, but in vain, all I found is an autosave.json file.

Thanks in advance

**Comment by francislavoie** [comment]
It gets output in the logs when CDP reloads your config based on container changes.

**Comment by Drillan767** [comment]
Hello, sorry for the delay but thanks for pointing out where to get it!

I finally managed to change the service to regerate a new Caddyfile, here is how it looks like:
```
{
  "apps": {
    "http": {
      "servers": {
        "srv0": {
          "listen": [
            ":443"
          ],
          "routes": [
            {
              "match": [
                {
                  "host": [
                    "subdomain1.somedomain.net"
                  ]
                }
              ],
              "handle": [
                {
                  "handler": "subroute",
                  "routes": [
                    {
                      "handle": [
                        {
                          "handler": "reverse_proxy",
                          "upstreams": [
                            {
                              "dial": "172.18.0.4:8081"
                            }
                          ]
                        }
                      ]
                    }
                  ]
                }
              ],
              "terminal": true
            },
            {
              "match": [
                {
                  "host": [
                    "subdomain2.domain.net"
                  ]
                }
              ],
              "handle": [
                {
                  "handler": "subroute",
                  "routes": [
                    {
                      "handle": [
                        {
                          "handler": "reverse_proxy",
                          "upstreams": [
                            {
                              "dial": "172.18.0.7:1234"
                            }
                          ]
                        }
                      ]
                    }
                  ]
                }
              ],
              "terminal": true
            },
            {
              "match": [
                {
                  "host": [
                    "subdomain3.domain.net"
                  ]
                }
              ],
              "handle": [
                {
                  "handler": "subroute",
                  "routes": [
                    {
                      "handle": [
                        {
                          "handler": "reverse_proxy",
                          "upstreams": [
                            {
                              "dial": "172.18.0.5:8080"
                            }
                          ]
                        }
                      ]
                    }
                  ]
                }
              ],
              "terminal": true
            }
          ]
        }
      }
    }
  }
}
``` 

I don't even see my custom configuration through the labels in it, so I guess it's not surprising it didn't work when I tried.

**Comment by Drillan767** [comment]
Hello, why was the issue closed? :sweat_smile: 
I still have the issue,I changed the labels to `caddy.log.format: console`, it was understood by Caddy but the output is still in JSON format and displays the `info` level of the logs

**Comment by masterkain** [comment]
I'm trying to hide these messages

```json
{"level":"info","ts":1630687314.413251,"logger":"docker-proxy","msg":"Skipping default Caddyfile because no path is set"}
{"level":"info","ts":1630687314.4133,"logger":"docker-proxy","msg":"Skipping swarm config caddyfiles because swarm is not available"}
{"level":"info","ts":1630687314.426478,"logger":"docker-proxy","msg":"Skipping swarm services because swarm is not available"}
```

and log normal http requests.

in docker-compose I have this:

```yaml
  caddy:
    labels:
      caddy.log: "{}"
      caddy.log.level: WARN
      caddy.log.include: docker-proxy
```

but it's not doing neither things, any hints?

**Comment by francislavoie** [comment]
> Hello, why was the issue closed? 😅

Because this issue is a question, and not an actionable bug or feature request.

>  I still have the issue,I changed the labels to caddy.log.format: console, it was understood by Caddy but the output is still in JSON format and displays the info level of the logs

@Drillan767 `console` format still contains JSON. You'll need to be more specific about your issue.

> I'm trying to hide these messages

Pretty sure those messages get emitted before the config is even loaded (because CDP's job is literally to generate a config from labels) so that's a bit of a chicken-and-egg problem, those can't really be configured.

That said, what you would probably need is actually something like this (but I don't think it'll actually work because of the above):

```
caddy.log: ""
caddy.log.exclude: docker-proxy
caddy.log: docker-proxy
caddy.log.include: docker-proxy
caddy.log.level: WARN
```

Basically, the main/default logger stops emitting `docker-proxy` logs, then a new logger is added to include those, but only at WARN and higher.
